test-poc/framework/src/main/java/org/keycloak/test/framework/injection/Supplier.java

```diff
@@ -41,6 +55,20 @@ default String getRef(S annotation) {
         return "";
     }
 
+    default String getRealmRef(S annotation) {
```

Comment:
All these methods that extract a value from an annotation could probably be done generically for every annotation attribute (ref, realmRef, config, lifecycle). I'll make a follow-up PR for it.

Reply:
yes, I concur, it is already 4 times in there :-) please add there another commit, since I've added realmRef and you config

Reply:
Sure, I will add it :slightly_smiling_face: